### ISSUE TITLE
Customizing HTML5 landing page..what am I doing wrong?

### ISSUE THREAD

**Issue by tarunb999** [bigbluebutton/greenlight#1360]
Hi,
I've been trying to get this done for more than a week but as I'm a total novice in Ruby, I just can't seem to get this done ..just need to change a few words in en.yml
So here's what I'm doing..

1. docker-compose down
cd ..
mv greenlight/ greenlight-old/

2. Forked the Greenlight repo to my github account.
3.     # git clone https://github.com/tarunb999/greenlight.git
cd greenlight
4. git status        _[got same msg as shown on docs page..]_
5. git remote add upstream https://github.com/bigbluebutton/greenlight.git
6. git fetch upstream
7. git checkout -b custom-changes upstream/v2

8. cp sample.env .env

9. docker run --rm bigbluebutton/greenlight:v2 bundle exec rake secret    
 _[pasted the key at end of SECRET_KEY_BASE= ... ]_
10. bbb-conf --secret
_[pasted the URL & SECRET at right places]_
11. docker run --rm --env-file .env bigbluebutton/greenlight:v2 bundle exec rake conf:check
_[ PASSED ..3 times, no error]_
12. cat ./greenlight.nginx | sudo tee /etc/bigbluebutton/nginx/greenlight.nginx
_[All looks good]_

13. systemctl restart nginx

14.  ./scripts/image_build.sh greenlocal release-v2
_[decide to name image name as greenlocal]_

15. nano docker-compose.yml   
Edited it to look:
 services:
  app:
    entrypoint: [bin/start]
    image: greenlocal:release-v2    

16. docker-compose up -d
17. bbb-conf --restart

Now my Greenlight is up & running.

18.  docker exec -it greenlocal bash
[This gives an error & says that greenlocal is not an image name]
So I instead run
docker exec -it greenlight-v2 bash

19. bash 4.4# vi config/locales/en.yml 
[Do my changes, save it]

20. exit the container & 
 docker-compose down
 docker-compose up -d
bbb-conf --restart

I don't see any change on my Landing page.. its all as it was ..When I come back and see en.yml the file has gone back to its original form.

I had even requisitioned another server , just to carry on these modifications without hurting my original install, but could not succeed.

Any advise pls as to how can I customize some wordings on en.yml and index.html.erb ??

**Comment by farhatahmad** [comment]
Hey @tarunb999 

You've got all of it down, just misunderstanding a step. 

> Now my Greenlight is up & running.

Everything up to this step is good and correct. Now, make the changes to your `config/locales/en.yml` file in your local machine (where you did the git clone and what not)  **NOT** in the greenlight-v2 container.
Then restart Greenlight 
```
docker-compose down
./scripts/image_build.sh greenlocal release-v2
docker-compose up -d
```

**Comment by tarunb999** [comment]
Ok.. so after Step 17 .. while still greenlight directory I did

# nano config/locales/en.yml
[edited en.yml and saved]

docker-compose down
./scripts/image_build.sh greenlocal release-v2
docker-compose up -d
bbb-conf --restart

But couldn't find any change to Landing Page .. when I checked back again ..en.yml was still having those changes but somehow they are not visible on Landing page.

**Comment by tarunb999** [comment]
Really sorry .. finally I deleted my previous git clone.. forked again..cloned again and then did all of these steps.
And YES... I got my new edited page !!!!

I'll try moving my old .env file and db file and see if I got my old accounts and recordings back !!
So I did..
cp ~/greenlight-old/.env ~/greenlight/.env
sudo cp -r ~/greenlight-old/db ~/greenlight/

docker-compose down
docker-compose up -d
bbb-conf --restart

Now getting 404 error :-( 

... and the battle continues !!!

**Comment by farhatahmad** [comment]
Do you have 2 db folders in ~/greenlight ?

**Comment by tarunb999** [comment]
root@e2e-58-97:~/greenlight# ls -al ~/greenlight
total 168
drwxr-xr-x 15 root root 4096 Apr 22 13:48 .
drwx------ 13 root root 4096 Apr 22 13:31 ..
-rw-r--r--  1 root root  286 Apr 22 13:31 .dockerignore
-rw-r--r--  1 root root 7908 Apr 22 14:05 .env
drwxr-xr-x  8 root root 4096 Apr 22 13:35 .git
-rw-r--r--  1 root root  858 Apr 22 13:31 .gitignore
-rw-r--r--  1 root root 1665 Apr 22 13:31 .gitlab-ci.yml
-rw-r--r--  1 root root  578 Apr 22 13:31 .rake_tasks~
-rw-r--r--  1 root root   53 Apr 22 13:31 .rspec
-rw-r--r--  1 root root 3426 Apr 22 13:31 .rubocop.yml
-rw-r--r--  1 root root    6 Apr 22 13:31 .ruby-version
-rw-r--r--  1 root root 1325 Apr 22 13:31 .travis.yml
-rw-r--r--  1 root root 1702 Apr 22 13:31 Dockerfile
-rw-r--r--  1 root root 3664 Apr 22 13:41 Gemfile
-rw-r--r--  1 root root 9818 Apr 22 13:35 Gemfile.lock
-rw-r--r--  1 root root 7652 Apr 22 13:31 LICENSE
-rw-r--r--  1 root root 2400 Apr 22 13:31 README.md
-rw-r--r--  1 root root  394 Apr 22 13:31 Rakefile
drwxr-xr-x 10 root root 4096 Apr 22 13:31 app
drwxr-xr-x  2 root root 4096 Apr 22 13:31 bin
-rw-r--r--  1 root root  934 Apr 22 13:31 cloudbuild-dev.yaml
-rw-r--r--  1 root root  906 Apr 22 13:31 cloudbuild.yaml
drwxr-xr-x  5 root root 4096 Apr 22 13:35 config
-rw-r--r--  1 root root  230 Apr 22 13:31 config.ru
drwxr-xr-x  4 root root 4096 Apr 22 13:49 db
-rw-r--r--  1 root root  824 Apr 22 13:47 docker-compose.yml
-rw-r--r--  1 root root  945 Apr 22 13:31 greenlight.nginx
drwxr-xr-x  4 root root 4096 Apr 22 13:31 lib
drwxr-xr-x  2 root root 4096 Apr 22 13:49 log
drwxr-xr-x  2 root root 4096 Apr 22 13:31 public
-rw-r--r--  1 root root 7681 Apr 22 13:35 sample.env
drwxr-xr-x  2 root root 4096 Apr 22 13:31 scripts
drwxr-xr-x  8 root root 4096 Apr 22 13:31 spec
drwxr-xr-x  3 root root 4096 Apr 22 13:31 test
drwxr-xr-x  2 root root 4096 Apr 22 13:31 tmp
drwxr-xr-x  3 root root 4096 Apr 22 13:31 vendor
-rw-r--r--  1 root root   86 Apr 22 13:31 yarn.lock

**Comment by tarunb999** [comment]
Well ..again I've now copied back my original greenlight directory to its rightful place and restarted docker. And my site  is running like last time without any changes.

Back to square one ..I'll try again trow !!!

**Comment by farhatahmad** [comment]
Yeah you have to be careful going back and forth. Your `.env` file stores your db password which gets set in your `docker-compose.yml`. 

There is a field at the bottom of the `docker-compose.yml` called `POSTGRES_PASSWORD`. 
There is also a field at the bottom of the `.env` called `DB_PASSWORD` 

Make sure these are set to the same value, if not, use the `POSTGRES_PASSWORD` for both

**Comment by tarunb999** [comment]
OK..so finally I got it .. I changed my HTML5 landing page.
I followed till step 17. Then followed these steps:

1.  #nano config/locales/en.yml
[edited en.yml and saved]

2.  docker-compose down
     ./scripts/image_build.sh greenlocal release-v2
     docker-compose up -d
     bbb-conf --restart

Saw my changes online.
To bring back my old accounts etc, followed these...

docker-compose down
cd ..
cp ~/greenlight-old/.env ~/greenlight/.env
cp -r ~/greenlight-old/db ~/greenlight/
docker-compose up -d
bbb-conf --restart

So finally, thanks Ahmad for all your patience and help. 
It would be better in the future if a page can be made by listing steps like above to enable editing the landing page.

Thanks again !!!
